### ISSUE TITLE
Result of prob different from approximate_distribution

### ISSUE THREAD

**Issue by naokityokoyama** [MaartenGr/BERTopic#1288]
I trained BERTopic with the following parameters:

#vectorizer
ctfidf  =  ClassTfidfTransformer()
bow  =  CountVectorizer(ngram_range=(1,3))

#Reduce dimensionality
umap_model = UMAP(n_neighbors=15, n_components=5, min_dist=0.0, metric='cosine', random_state=42)

#Cluster reduced embeddings
hdbscan_model = HDBSCAN(min_cluster_size=15, metric='euclidean', cluster_selection_method='eom', prediction_data=True)

#model
embedding_model = 'all-MiniLM-L6-v2'

model = BERTopic(language='Portuguese', 
                 embedding_model = embedding_model, 
                 umap_model = umap_model,
                 hdbscan_model = hdbscan_model,
                 vectorizer_model = bow,
                 ctfidf_model=ctfidf,
                 calculate_probabilities=True,
                 nr_topics = 11, 
                 verbose=True)

topics, prob = model.fit_transform(corpus)
set(topics) {-1, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9}

topic_distr, _ = model.approximate_distribution(corpus)

topics[2]
0

prob[2]
array([**0.81462017**, 0.00272969, 0.00171482, 0.00218249, 0.00305799,
       0.02503531, 0.00912715, 0.00192635, 0.00169254, 0.00191875])

topic_distr[2]
array([0.        , 0.        , 0.        , 0.        , 0.        ,
       **0.54159006**, 0.45840994, 0.        , 0.        , 0.        ])

topic 2 has a value of 0
but if you look at the probability, in fact, 0 is the one that has the highest but if you look at the topic_distr, the highest value is the 5th position

why?

I'm using approximate_distribution to return the probability of topics in the same sentence

**Comment by MaartenGr** [comment]
That is to be expected since `.approximate_distribution` is exactly that, an approximation of probabilities and not the exact same probabilities that you will find with `.fit_transform`. More specifically, the probabilities from `.fit_transform` are generated through the clustering algorithm whereas `.approximate_distribution` is a technique independent of the clustering algorithm that you use. That means that `.approximate_distribution` can be used even when you have clustering algorithms, like k-Means, that do not perform soft-clustering directly. 

You can find more about [`.approximate_distribution`](https://maartengr.github.io/BERTopic/getting_started/distribution/distribution.html) in the documentation.

**Comment by MaartenGr** [comment]
Yes, you can find different probabilities using `.approximate_distribution`. Try out both methods to see which one fits your use case best.
